internal/consensus/wal_test.go

```diff
@@ -105,14 +105,14 @@ func TestWALWrite(t *testing.T) {
 	walDir := t.TempDir()
 	walFile := filepath.Join(walDir, "wal")
 
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
 	wal, err := NewWAL(log.TestingLogger(), walFile)
 	require.NoError(t, err)
-	err = wal.Start()
+	err = wal.Start(ctx)
 	require.NoError(t, err)
 	t.Cleanup(func() {
-		if err := wal.Stop(); err != nil {
-			t.Error(err)
-		}
 		// wait for the wal to finish shutting down so we
 		// can safely remove the directory
 		wal.Wait()
```

Comment:
```suggestion
```

Reply:
This suggestion is incomplete because part of the affected code is outside the diff and GitHub hates fun.